### PR TITLE
[tools] Fix publishing with node 20

### PR DESCRIPTION
**Pull request by wschurman**
# Why

We're hitting https://github.com/npm/cli/issues/7354 in our publish script now that we use node 20.

# How

Apply the workaround for now.

Also show a more full error when this fails for future failures if any.

# Test Plan

`et publish --canary --dry` and see it no longer gets stuck.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by byCedric — APPROVED** [review]

File: tools/src/publish-packages/tasks/packPackageToTarball.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export const packPackageToTarball = new Task<TaskArgs>(
           return;
         } catch (error) {
           step.fail();
-          logger.error(error.stderr);
+          logger.error(error);
```
Comment (byCedric) [inline tools/src/publish-packages/tasks/packPackageToTarball.ts:34]: Was this change intentional? I think we still only want to show the actual error message, right? Or the whole object?

Comment (wschurman) [reply tools/src/publish-packages/tasks/packPackageToTarball.ts:34]: Yes, intentional. It was showing `undefined` before since stderr didn't always exist I think, or at least for this bug. Changing this to print the whole error shows the full output.
